src/main/java/org/zalando/nakadi/validation/schema/diff/CombinedSchemaDiff.java

```diff
@@ -17,8 +17,7 @@
     static void recursiveCheck(final CombinedSchema combinedSchemaOriginal, final CombinedSchema combinedSchemaUpdate,
                                        final Stack<String> jsonPath,
                                        final List<SchemaChange> changes) {
-        if(!(combinedSchemaOriginal.getSubschemas().containsAll(combinedSchemaUpdate.getSubschemas())
-                && combinedSchemaUpdate.getSubschemas().containsAll(combinedSchemaOriginal.getSubschemas()))) {
+        if(combinedSchemaOriginal.getSubschemas().size() != combinedSchemaUpdate.getSubschemas().size()) {
```

Comment:
Yes. They could be reordered. I considered performing some kind of "json sorting" before comparing. But for the sake of simplicity it doesn't support reordering of elements in arrays.